lib/formValidation.js

```diff
@@ -0,0 +1,52 @@
+export const name = (value) => /^[a-zA-Z]{4,}(?: [a-zA-Z]+){0,2}$/.test(value);
+export const password = (value) =>
+  /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/.test(value);
+export const email = (value) =>
+  /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value);
+//export const requiredField = value =>  (value || typeof value === 'number' ? undefined : 'Required')
+
+export const validateFunc = (value) => {
+  const error = {};
+  if (!value.firstname) {
```

Comment:
can we divide this functionality into smaller function and instead of having validations on the first name can we have a validation based on the field type as this would make these validations more generic. and please create a file for these error values.